xbmc/guilib/GUIWindowManager.h

```diff
@@ -227,6 +231,7 @@ class CGUIWindowManager : public KODI::MESSAGING::IMessageTarget
 
   typedef std::map<int, CGUIWindow *> WindowMap;
   WindowMap m_mapWindows;
+  std::unordered_map<std::type_index, CGUIWindow*> m_mapWindowTypes;
```

Comment:
I use an unordered_map here to optimize map lookups. We don't need an ordered map and only insert/remove items on start and shutdown of the app.

Four years ago a cache `CGUIWindowManagerIdCache` was introduced to optimize the map lookup of std::map in `GetWindow(int id)`. Not sure if he checked the performance of an unordered_map.

https://github.com/xbmc/xbmc/commit/98b4f9c083fc7c6ec1549228ac59e9d3b6ad9bbc